yt_dlp/utils/_utils.py

```diff
@@ -2010,13 +2010,18 @@ def str_or_none(v, default=None):
     return default if v is None else str(v)
 
 
-def str_to_int(int_str):
+def str_to_int(int_str, *, dot_decimal=False):
```

Comment:
There is a fine old phrase "putting lipstick on a pig". However things like `locale.atoi()` are even piggier, so I can see that improving `str_to_int()` might be a good thing.

[International standards, though little practised, suggest using a single space rather than `.` or `,`. Other cases might be `_`, `'`](https://en.wikipedia.org/wiki/Decimal_separator#Digit_grouping). Maybe, instead of a binary choice,
```suggestion
def str_to_int(int_str, *, separator=',':
```
or
```suggestion
def str_to_int(int_str, *, thousands_separator=',':
```
and so on.

The thousands separator and the decimal point must differ, so the to-be-converted string will be the maximal length string comprising digits and the thousands separator, possibly with an initial `+`, or more specifically matching `r'(?A)\+?\d+(?:%s\d+)+' % separator`.

Reply:
Wouldn't this be a breaking change? Both `.` and `,` are currently removed as separators

Reply:
How about setting ('+', ',') as the default characters to ignore?
```python
def str_to_int(int_str, *, dot_decimal=False, ignored_chars=('+', ',')):
    ...
    int_str = re.sub(rf'[{re.escape("".join(
        variadic(ignored_chars, allowed_types=str)))}]', '', int_str)
```

Reply:
I think maybe the simplest way do this is with a regex kwarg
```py
def new_str_to_int(int_str, *, ignore_pattern=r'[,.+]'):
    if isinstance(int_str, int):
        return int_str
    if not isinstance(int_str, str):
        return None
    int_str = re.sub(ignore_pattern, '', int_str)
    return int_or_none(float_or_none(int_str))
```

Reply:
quick question: in the ignore pattern [+,.], is the + intended to strip only a leading plus sign, or all plus signs in the string? If it’s only the former, isn’t that redundant since Python’s int and float already accept a leading plus?
also I can’t think of any case where removing non-leading plus sign and concatenating the digits makes sense.

Reply:
I thik the suggested solution doesnt quite work in our favor. You could not express german string notification with this, eg. `123.456,78`. I think explicit string transformations might be better here, and behavior shuold be left unchanged?

Maybe with a note explaining `str_to_int` is no good?

Reply:
German formats aren’t supported by `float_or_none` or similar functions; pre-normalization is required and falls outside `str_to_int`’s scope

```python
v.replace('.', '').replace(',', '.') -> str_to_int(v, ignore_pattern='')
v.translate(str.maketrans('.,', ',.')) -> str_to_int(v, ignore_pattern=',')
```

Reply:
Im saying that, if prenormalization is required anyways for other locales, there is not much use in a variable regex approach, the boolean would be fine.

But at that point you could also compose `float_or_none` and `int_or_none`; the only benefit this would give over that was if it were to also apply `partial_application`.

Long story short: imo boolean argument + `partial_application` decorator would be the way

Reply:
Prenormalisation isn't "normally" required, since [the default stripped characters are `+,.`](https://github.com/yt-dlp/yt-dlp/pull/13212/files#r2094632659). The last 2 cover the most frequent thousand separators, but not the problem that the PR is addressing, converting a non-negative numeric string with thousands separators that possibly also includes a decimal fraction part to an `int`, which is a reasonable case especially (only?) when the fraction part is 0. Unlike, say, `20,000.0`, it's not possible to disambiguate `20.000` without knowing either what the thousands separator is, or what the decimal point is, the two being mutually exclusive.

Then:
* if a thousands separator is detected or specified that is `.` or `,`, any decimal point must be the other one
* if any other thousands separator is detected or specified, any decimal point might be either one (since we are **not** performing strict `locale` processing).

It seems better to start from the thousands separator since that has more possible values than decimal points. 

With an option to allow automatic handling of decimal fractions, consider what result should be returned in these cases. Is any decimal fraction to be ignored, or to be used for rounding, or only allowed to be 0?
* `20 000.0`
* `2 000.141` (`Duration: 2 000.141`, `Views: 2 000.141 viewers were anonymous.`)
*  ... perm thousands separators and decimal points in the above.
Tricky (a hypothetical site using `.` as thousands separator and also omitting a space after full-stop):?
`Views: 2.000.141 viewers were anonymous.`

I'd say that, if adding a decimal point or thousands separator option, any decimal fraction must be 0, since this function is for parsing an integer string, not converting a generic decimal literal to `int`.

But that does raise the question as to what actual requirement prompts the change.

Reply:
I believe the key points are:
- retain the existing behavior as the default
- add an option to customize the filter pattern (also enabling decimal dropping)
- apply `@partial_application`
- remove the vestigial handling of `+`

(since `float_or_none`/`int_or_none` only accept standard formats, a custom decimal-separator flag adds little value)

Ultimately, how about something like this?
```python
@partial_application
def str_to_int(v, ignore=r'[,.]'):
    if isinstance(v, int):
        return v
    if not isinstance(v, str):
        return None

    return int_or_none(float_or_none(re.sub(ignore, '', v)))
```